src/Features/CSharp/Portable/IntroduceUsingStatement/CSharpIntroduceUsingStatementCodeRefactoringProvider.cs

```diff
@@ -61,5 +59,23 @@ protected override StatementSyntax CreateUsingStatement(LocalDeclarationStatemen
                 expression: null, // Declaration already has equals token and expression
                 SyntaxFactory.Token(SyntaxKind.CloseParenToken).WithTrailingTrivia(declarationStatement.GetTrailingTrivia()),
                 statement: SyntaxFactory.Block(statementsToSurround));
+
+        protected override bool TryCreateUsingLocalDeclaration(
+            ParseOptions options,
+            LocalDeclarationStatementSyntax declarationStatement,
+            [NotNullWhen(true)] out LocalDeclarationStatementSyntax? usingDeclarationStatement)
+        {
+            usingDeclarationStatement = null;
+            if (declarationStatement.Parent is not BlockSyntax ||
```

Comment:
Why the block statement check? Seems like it would just prevent this from working in top level statements, and perhaps could be a little more specific for the scenario we're trying to avoid?

Reply:
ah good point.  this was really about not doing this in a switch-case-section (as those don't allow these unbounded using ranges).  But i should just check for that, instead of this specific check).

Reply:
switched.